gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/CollectionUtil.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.tinkerpop.gremlin.util;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class CollectionUtil {
```

Comment:
there is a `org.apache.tinkerpop.gremlin.util.tools` with a `CollectionFactory`. should we move the functions there to this class and remove `CollectionFactory`? i think this is the better named class. may as well get rid of "tools" package and just move `MultiMap` up a level to this package. please `final` variables in this class and add unit tests for `clone()` - any reason the method uses a capital "C" and not lower case?

Reply:
consolidated in https://github.com/apache/tinkerpop/pull/2087/commits/9b6991dd123267ae5ffa536cb608e81d4c999aad